### PR TITLE
Support creation of influxDB instance

### LINKED ISSUE #4114: Stand up a separate influxdb server instance from metrics.solana.com
metrics.solana.com is doing too many things.  When we flood the influxdb instance on m.s.c, the stability of other systems are affected.

Let's setup a VM instance that's sole job is to run influxdb.   The creation of the instance should be fully scripted, probably under `net/`, because:
a. the VM instance should be easy to dispose of and recreate if we need to
b. by using `net/`, we can be cloud provider agnostic.  Maybe we'll use AWS for the influxdb database today, Azure tomorrow
c. in the future we may want to spin up an influxdb instance only for a particular testnet

### PR DESCRIPTION
**Pull request by danpaul000**
#### Problem

We need a way to stand up a new instance as needed just to run influxDB to offload metrics-solana-com and the cloud service

#### Summary of Changes

Extend gce.sh to create/config/delete the instance by adding the "-i" option

Fixes #4114

### CODE REVIEW

File: net/gce.sh [hunk 1]
```diff
@@ -650,10 +677,94 @@ EOF
...
+
+  rm -rf "$sshPrivateKey"{,.pub}
+
+  # Note: using rsa because |aws ec2 import-key-pair| seems to fail for ecdsa
+  ssh-keygen -t rsa -N '' -f "$sshPrivateKey"
+
+
+  declare startupScript="$netConfigDir"/instance-startup-script.sh
+  cat > "$startupScript" <<EOF
+#!/usr/bin/env bash
```
Comment (danpaul000) [inline net/gce.sh]: Not sure if any of this startup script is needed

File: net/init-metrics.sh [hunk 2]
```diff
@@ -30,7 +30,8 @@ loadConfigFile
 
 useEnv=false
 delete=false
-host="https://metrics.solana.com:8086"
```
Comment (danpaul000) [inline net/init-metrics.sh]: Add a way to dynamically specify the hostname or IP of the metrics server

**Comment by danpaul000** [comment]
Ideally, after VM creation, steps on the new instance would be something like:
```sudo snap install docker
git clone metrics-solana-com
~/metrics-solana-com/start.sh```

Looks like the https certificate generation was done manually for m-s-c. As well as some directory creation.  Should these steps be put into a startup script for this flavor of VM?

**Comment by mvines** [comment]
Oh so I wasn't thinking we'd plumb this through `./gce.sh`.  Instead build it into `./init-metrics.sh` or a new `./metrics-host.sh` (`./metrics-host.sh create`/`./metrics-host.sh delete`) that uses the underlying cloud provider scripts like `./gce.sh` does.  Because:
* We may want to keep the metrics server up after tearing down the potentially expensive testnet
* We may want to share a metrics server cross multiple testnets

**Comment by danpaul000** [comment]
> Oh so I wasn't thinking we'd plumb this through `./gce.sh`. Instead build it into `./init-metrics.sh` or a new `./metrics-host.sh` (`./metrics-host.sh create`/`./metrics-host.sh delete`) that uses the underlying cloud provider scripts like `./gce.sh` does. Because:
> 
>     * We may want to keep the metrics server up after tearing down the potentially expensive testnet
> 
>     * We may want to share a metrics server cross multiple testnets

As written, the influx instance is created/torn down independent of any testnet.  It just seemed like the path of least resistance to make use of existing logic to bring up new instances.  But I can pop it out to its own script if you want.  Creating the VM is the less challenging part, I am still fighting to actually get a useful DB running on the host once its online.

**Comment by mvines** [comment]
Ah, `-i` . I see.  Hmm yeah I feel like a separate script will ultimately be much more maintainable.   But yeah probably get it working first in the existing structure :)

**Pull request closed without merging**